packages/task/src/browser/quick-open-task.ts

```diff
@@ -196,25 +199,39 @@ export class QuickOpenTask implements QuickOpenModel, QuickOpenHandler {
     attach(): void {
         this.items = [];
         this.actionProvider = undefined;
-
+        const isMulti: boolean = this.workspaceService.isMultiRootWorkspaceOpened;
         this.taskService.getRunningTasks().then(tasks => {
             if (!tasks.length) {
                 this.items.push(new QuickOpenItem({
                     label: 'No tasks found',
                     run: (_mode: QuickOpenMode): boolean => false
                 }));
-            }
-            for (const task of tasks) {
-                // can only attach to terminal processes, so only list those
-                if (task.terminalId) {
-                    this.items.push(
-                        new TaskAttachQuickOpenItem(
+            } else {
+                tasks.forEach((task: TaskInfo) => {
+                    // can only attach to terminal processes, so only list those
+                    if (task.terminalId) {
+                        this.items.push(new RunningTaskQuickOpenItem(
                             task,
-                            this.getRunningTaskLabel(task),
-                            this.taskService
-                        )
-                    );
-                }
+                            this.taskService,
+                            this.taskNameResolver,
+                            this.taskSourceResolver,
+                            this.taskDefinitionRegistry,
+                            this.labelProvider,
+                            isMulti,
+                            {
+                                run: (mode: QuickOpenMode): boolean => {
+                                    if (mode !== QuickOpenMode.OPEN) {
+                                        return false;
+                                    }
+                                    if (task.terminalId) {
```

Comment:
is it required?
If I understand correctly you have this check here https://github.com/eclipse-theia/theia/pull/7392/files#diff-ea22f8aecaf89629bb85777e1be8833cR212

Reply:
yep you are right. I resolved your comments